internal/controller/genericprovider_controller.go

```diff
@@ -206,7 +206,7 @@ func (r *GenericProviderReconciler) reconcile(ctx context.Context, provider gene
 
 		if !res.IsZero() || err != nil {
 			// the steps are sequential, so we must be complete before progressing.
-			return res, err
+			return reconcile.Result{}, err
```

Comment:
If any of the phases being looped over returned a result with RequeueAfter, then that would be ignored with this change.

Reply:
Yes, I know.
This is specially for the catching secondary events from the CoreProvider, There wasn't problem with it.
The single phase which is looking for the repeated Requeue is `preflightChecks`. 
It seems to code should be refactored a bit, but not for now.

Reply:
What I wanted to highlight was that it should work as you intended even without this change because it will return the `res` that is returned from `preflightChecks` (or any other phase).

Reply:
Yes, agree.
But in this case, we will get an additional reconcile cycles  while waiting CoreProvider to be ready instead of just catching event from it.